opendbc/car/toyota/carcontroller.py

```diff
@@ -154,10 +167,36 @@ def update(self, CC, CS, now_nanos):
 
       net_acceleration_request = actuators.accel + accel_due_to_pitch
 
+      jerk1 = float(np.mean(np.abs(np.diff(list(self.pcm_accel_net_deque)[-50:])))) * 100
+      jerk2 = abs(self.pcm_accel_net_deque[-1] - self.pcm_accel_net_deque[-50]) * 2
+      jerk3 = abs(self.pcm_accel_net_filter.x - ((self.pcm_accel_net_deque[-1] - accel_due_to_pitch) - CS.out.aEgo))
+      # self.debug3 = jerk1
+      # self.debug4 = jerk3
+      if CS.out.standstill or stopping:# or (abs(self.pcm_accel_net_deque[-1] - self.pcm_accel_net_deque[-10]) * 10) > 1.5:
+        self.pcm_accel_net_filter.x = 0
+        # self.pid.reset()
+        new_pcm_accel_net = CS.pcm_accel_net
+      else:
+        # TODO: decide a good value for past pcm accel net
+        # find closest value to aEgo in pcm_accel_net_deque
+        arr = np.array(self.pcm_accel_net_deque) - accel_due_to_pitch
+        idx = (np.abs(arr - CS.out.aEgo)).argmin()
+        self.cur_idx = clip(idx, self.cur_idx - 1/3, self.cur_idx + 1/3)
+        self.debug3 = round(float(self.cur_idx))
+        print(self.cur_idx)
```

Comment:
We need this dynamic indexing to solve innacuracies under high jerk. Here is assuming a 0.4s offset:

![image](https://github.com/user-attachments/assets/2f387c73-d139-42bb-b4d1-7c6426055059)

Here is the dynamic indexing:

![image](https://github.com/user-attachments/assets/190de03f-f421-4aba-ba12-39d8ba52a1c4)

Reply:
And here is both gas and brake w/ dynamic indexing (from maneuver report):

![image](https://github.com/user-attachments/assets/33cdbc22-aba8-448a-b213-667fb853b04a)

Basically, a hard-coded offset that worked for one case/car didn't work for another